src/main/java/de/tum/in/www1/artemis/service/TutorLeaderboardService.java

```diff
@@ -76,27 +78,36 @@ public TutorLeaderboardService(ResultRepository resultRepository, ComplaintRepos
         log.info("Finished >>complaintRepository.findTutorLeaderboardAnsweredMoreFeedbackRequestsByCourseId<< call for course " + course.getId() + " in " + (end - start) + "ms");
 
         return aggregateTutorLeaderboardData(tutors, tutorLeaderboardAssessments, tutorLeaderboardComplaints, tutorLeaderboardMoreFeedbackRequests,
-                tutorLeaderboardComplaintResponses, tutorLeaderboardAnsweredMoreFeedbackRequests);
+                tutorLeaderboardComplaintResponses, tutorLeaderboardAnsweredMoreFeedbackRequests, false);
     }
 
     /**
      * Returns tutor leaderboards for the specified course.
      *
      * @param course - course for which leaderboard is fetched
-     * @param exam   - the exam for which the leaderboard will be fetched TODO
+     * @param exam   - the exam for which the leaderboard will be fetched
      * @return list of tutor leaderboard objects
      */
     public List<TutorLeaderboardDTO> getExamLeaderboard(Course course, Exam exam) {
 
         List<User> tutors = userRepository.getTutors(course);
-        // TODO: get the exam leaderboard. We do not want to use the existing s. We do not yet support the exam leaderboard
-        // TODO: remove as soon as the above calls work
-        List<TutorLeaderboardAssessment> tutorLeaderboardAssessments = new ArrayList<>();
-        List<TutorLeaderboardComplaints> tutorLeaderboardComplaints = new ArrayList<>();
-        List<TutorLeaderboardComplaintResponses> tutorLeaderboardComplaintResponses = new ArrayList<>();
+        String groupName = course.getTeachingAssistantGroupName();
+
+        long start = System.currentTimeMillis();
+        List<TutorLeaderboardAssessments> tutorLeaderboardAssessments = resultRepository.findTutorLeaderboardAssessmentByExamId(exam.getId());
+        long end = System.currentTimeMillis();
+        log.info("Finished >>resultRepository.findTutorLeaderboardAssessmentByExamId<< call for exercise " + exam.getId() + " in " + (end - start) + "ms");
+        start = System.currentTimeMillis();
+        List<TutorLeaderboardComplaints> tutorLeaderboardComplaints = complaintRepository.findTutorLeaderboardComplaintsByExamId(groupName, exam.getId());
+        end = System.currentTimeMillis();
+        log.info("Finished >>complaintRepository.findTutorLeaderboardComplaintsByExamId<< call for exercise " + exam.getId() + " in " + (end - start) + "ms");
```

Comment:
Are these logs intentional?

Reply:
yes they are, we want to test the performance of those queries, even (or especially) in production.